SimTracker/TrackerHitAssociation/plugins/ClusterTPAssociationHeterogeneous.cc

```diff
@@ -44,9 +45,8 @@
 
 #include "ClusterSLOnGPU.h"
 
-class ClusterTPAssociationHeterogeneous : public HeterogeneousEDProducer<heterogeneous::HeterogeneousDevices<
-          heterogeneous::GPUCuda, heterogeneous::CPU>>
-{
+class ClusterTPAssociationHeterogeneous
+    : public HeterogeneousEDProducer<heterogeneous::HeterogeneousDevices<heterogeneous::GPUCuda, heterogeneous::CPU>> {
```

Comment:
`ClusterTPAssociationHeterogeneous` seems to stay `HeterogeneousEDProducer`, so the title of this PR is misleading.

By cursory look this PR introduces the `TrackingRecHit2DCUDA`, and migrates the `CklusterTPAssociationHeterogeneous` do use that. The hit-producing code `SiPixelRecHitHeterogeneous` appears to still produce `HeterogeneousProduct`, so there is nothing that would produce `TrackingRecHit2DCUDA`.

Should this PR also include the migration of `SiPixelRecHitHeterogeneous` to the new framework?